membrane_curvature/tests/test_membrane_curvature.py

```diff
@@ -297,13 +350,238 @@ def test_analysis_mean(self, universe):
         avg_mean = mc.results.average_mean
         assert_almost_equal(avg_mean, expected_mean)
 
-
-    def test_analysis_mean_default_selection(self, universe):
+    # test using wrap=False with test grofile
+    def test_analysis_mean_no_wrap(self, universe):
         expected_mean = np.array([[7.50000000e+00,  1.33985392e-01,  2.77315457e-04],
                                   [-2.77315457e-04, -3.53944270e-01, -7.50000000e+00],
                                   [-2.77315457e-04, -5.01100068e-01, -7.50000000e+00]])
         mc = MembraneCurvature(universe,
+                               select='name PO4',
+                               n_x_bins=3,
+                               n_y_bins=3,
+                               wrap=False).run()
+        avg_mean = mc.results.average_mean
+        assert_almost_equal(avg_mean, expected_mean)
+
+    # test using dummy Universe with atoms out of boounds
+    # with wrap=True (default)
+    #   +-----------+          +-----------+
+    #   |   | 7 | 8 | 9        | 9 | 7 | 8 |
+    #   +-----------+          +-----------+
+    # 4 | 5 | 6 |   |   --->   | 5 | 6 | 4 |
+    #   +-----------+          +-----------+
+    #   |   | 1 | 2 | 3        | 3 | 1 | 2 |
+    #   +-----------+          +-----------+
+    #
+    # test surface in universe with atoms out of bounds in x
+    def test_analysis_get_z_surface_wrap(self, universe_dummy_wrap, dummy_surface):
+        x_bin = y_bin = 3
+        mc = MembraneCurvature(universe_dummy_wrap,
+                               n_x_bins=x_bin,
+                               n_y_bins=y_bin).run()
+        avg_surface = mc.results.average_z_surface
+        assert_almost_equal(avg_surface, dummy_surface)
+
+    # test surface in universe with atoms out of bounds in x and y
+    def test_analysis_get_z_surface_wrap_xy(self, universe_dummy_wrap_xy, dummy_surface):
+        x_bin = y_bin = 3
+        mc = MembraneCurvature(universe_dummy_wrap_xy,
+                               n_x_bins=x_bin,
+                               n_y_bins=y_bin).run()
+        avg_surface = mc.results.average_z_surface
+        assert_almost_equal(avg_surface, dummy_surface)
+
+    # test mean curvature
+    def test_analysis_mean_wrap(self, universe_dummy_wrap, dummy_surface):
+        x_bin = y_bin = 3
+        mc = MembraneCurvature(universe_dummy_wrap,
+                               n_x_bins=x_bin,
+                               n_y_bins=y_bin).run()
```

Comment:
Seeing as you repeat the same analysis over and over, a fixture might be a good choice to save time.

```python
@pytest.fixture()
def wrapped_curvature_of_unwrapped_universe(universe_dummy_wrap):
	return MembraneCurvature(universe_dummy_wrap,
                               n_x_bins=3,
                               n_y_bins=3).run()

def test_analysis_mean_wrap(self, wrapped_curvature_of_unwrapped_universe, dummy_surface):
	avg_mean = wrapped_curvature_of_unwrapped_universe.results.average_mean
    expected_mean = mean_curvature(dummy_surface)
    assert_almost_equal(avg_mean, expected_mean)
```